src/Avalonia.Controls/TopLevel.cs

```diff
@@ -232,6 +235,18 @@ protected bool AutoSizing
         }
 
         /// <inheritdoc/>
+        IRenderTarget IRenderRoot.CreateRenderTarget()
+        {
+            return _renderInterface.CreateRenderer(PlatformImpl.Handle);
```

Comment:
Right, that's what I was going for with this. `IRenderRoot` is implemented by `TopLevel` and that can choose how to handle it; including passing the request to `ITopLevelImpl`. 

If I understand you correctly you're talking about losing `IRenderTarget` too, and just having something like `IRenderRoot.CreateDrawingContext` (which TopLevel will choose to pass to `ITopLevelImpl.CreateRenderTarget`)? If so, I think that's a good idea.